### PR TITLE
Add load test pipeline

### PR DESCRIPTION
**Pull request by ingwarsw**
Add load tests to see time difference between tar/registry on different image sizes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -690,6 +690,53 @@ jobs:
...
         run: earthly --use-inline-cache +for-linux
+      - name: Set EARTHLY_VERSION_FLAG_OVERRIDES env
+        run: |-
+          set -euo pipefail
+          EARTHLY_VERSION_FLAG_OVERRIDES="$(tr -d '\n' < .earthly_version_flag_overrides)"
+          echo "EARTHLY_VERSION_FLAG_OVERRIDES=$EARTHLY_VERSION_FLAG_OVERRIDES" >> "$GITHUB_ENV"
+      - name: Enable local registry-based exporter
+        run: ./build/linux/amd64/earthly config global.local_registry_host 'tcp://127.0.0.1:8371'
+      - name: Configure Earthly Conversion Parallelism
+        run: ./build/linux/amd64/earthly config global.conversion_parallelism 5
```
Comment (vladaionescu) [inline .github/workflows/ci.yml:701]: These are no longer necessary as they are both enabled by default now.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -690,6 +690,53 @@ jobs:
...
+      - name: run export tests
+        run: env earthly=./build/linux/amd64/earthly scripts/tests/export.sh
+      - name: Buildkit logs (runs on failure)
+        run: docker logs earthly-buildkitd
+        if: ${{ failure() }}
+
+  load-imagees-test:
```
Comment (vladaionescu) [inline .github/workflows/ci.yml:708]:
```suggestion
  load-images-test:
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -705,6 +705,48 @@ jobs:
...
+        if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository
+      - name: Build latest earthly using released earthly
+        run: earthly --use-inline-cache +for-linux
+      - name: Set EARTHLY_VERSION_FLAG_OVERRIDES env
+        run: |-
+            set -euo pipefail
+            EARTHLY_VERSION_FLAG_OVERRIDES="$(tr -d '\n' < .earthly_version_flag_overrides)"
+            echo "EARTHLY_VERSION_FLAG_OVERRIDES=$EARTHLY_VERSION_FLAG_OVERRIDES" >> "$GITHUB_ENV"
+      - name: Run load tests
+        run: ./build/linux/amd64/earthly -P ./tests/${{ matrix.type }}/+all --TEST_IMAGE=${{ matrix.test_image }}M --earthly=$(realpath ./build/linux/amd64/earthly)
```
Comment (alexcb) [inline .github/workflows/ci.yml:745]: looks like a `M` suffix slipped in.

```suggestion
        run: ./build/linux/amd64/earthly -P ./tests/${{ matrix.type }}/+all --TEST_IMAGE=${{ matrix.test_image }} --earthly=$(realpath ./build/linux/amd64/earthly)
```

File: tests/with-docker-registry/Earthfile [hunk 4]
```diff
@@ -37,23 +41,63 @@ docker-load-test:
...
-            --pull hello-world \
-            --load +a-test-image
-        RUN docker run test-abc-img:xyz && \
-            docker run hello-world
+        --load=(+another-test-image --INDEX=1) \
+        --load=(+another-test-image --INDEX=2) \
+        --load=(+another-test-image --INDEX=3) \
+        --load=(+another-test-image --INDEX=4) \
+        --load=(+another-test-image --INDEX=5)
+        RUN docker run --rm another-test-img:i1 && \
```
Comment (alexcb) [inline tests/with-docker-registry/Earthfile:89]: Is it possbile to add back in the grep which is used to ensure the `INDEX` arg gets correctly propigated into the container that's being built and run?
e.g.

```suggestion
        RUN docker run --rm another-test-img:i1 | grep "test img 1" && \
```

(and `grep 2`,...`grep 5`)

**Comment by alexcb** [comment]
Thanks for the PR @ingwarsw , sorry about the delay in taking a look at this.

I'm curious if it should be more of a manually triggered job or not (depending on how much time it adds to our CI), I had a go at trying it out by shifting the load test into a seperate `load-test.yml` github action config and kicked off a build under:
 https://github.com/alexcb/earthly/runs/6870605343?check_suite_focus=true

and it looks like the `aergus/latex` test might be pushing it a little too much load:

```
./t/with-docker-registry+docker-load-test *failed* | failed to register layer: ApplyLayer exit status 1 stdout:  stderr: write /usr/share/asymptote/geometry.asy: no space left on device
```

were you able to run this via github actions, or were you just testing it locally?

I tried disabling all of our other tests in my [own branch](https://github.com/alexcb/earthly/commit/9abcd429f40e8961d691edde36ccc29e41aefd23) to help me iterate faster. (perhaps my own personal branch is limited to less hard drive space compared to your testing environment?)

**Pull request closed without merging**